libsolidity/ast/Types.cpp

```diff
@@ -3406,36 +3424,20 @@ MemberList::MemberMap TypeType::nativeMembers(ContractDefinition const* _current
 	if (m_actualType->category() == Category::Contract)
 	{
 		ContractDefinition const& contract = dynamic_cast<ContractType const&>(*m_actualType).contractDefinition();
-		bool isBase = false;
-		if (_currentScope != nullptr)
-		{
-			auto const& currentBases = _currentScope->annotation().linearizedBaseContracts;
-			isBase = (find(currentBases.begin(), currentBases.end(), &contract) != currentBases.end());
-		}
-		if (isBase)
-		{
-			// We are accessing the type of a base contract, so add all public and protected
-			// members. Note that this does not add inherited functions on purpose.
-			for (Declaration const* decl: contract.inheritableMembers())
-				members.emplace_back(decl->name(), decl->type(), decl);
-		}
-		else
+		bool inDerivingScope = _currentScope && _currentScope->derives(contract);
+
+		for (auto const* declaration: contract.declarations())
 		{
-			bool inLibrary = contract.isLibrary();
-			for (FunctionDefinition const* function: contract.definedFunctions())
-				if (
-					(inLibrary && function->isVisibleAsLibraryMember()) ||
-					(!inLibrary && function->isPartOfExternalInterface())
-				)
-					members.emplace_back(
-						function->name(),
-						FunctionType(*function).asCallableFunction(inLibrary),
-						function
-					);
-			for (auto const& stru: contract.definedStructs())
-				members.emplace_back(stru->name(), stru->type(), stru);
-			for (auto const& enu: contract.definedEnums())
-				members.emplace_back(enu->name(), enu->type(), enu);
+			if (dynamic_cast<ModifierDefinition const*>(declaration))
```

Comment:
Can you explain this?

Reply:
Modifiers are not accessible via contract name...